File: src/main/java/tk/hiddenname/smarthome/model/task/processing/objects/SetPwmSignalObject.java
package tk.hiddenname.smarthome.model.task.processing.objects;

import com.fasterxml.jackson.databind.PropertyNamingStrategy;
import com.fasterxml.jackson.databind.annotation.JsonNaming;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Table;
import javax.validation.constraints.NotNull;

@Data
@Entity
@NoArgsConstructor
@EqualsAndHashCode(callSuper = true)
@Table(name = "processing_set_pwm_signal")
@JsonNaming(PropertyNamingStrategy.SnakeCaseStrategy.class)
public class SetPwmSignalObject extends ProcessingObject {

    @NotNull
    @Column(name = "device_id", nullable = false)
    private Long deviceId;

    @Column(nullable = false)
    private int delay = 0;

    @NotNull
    @Column(nullable = false)
    private int targetSignal;
}

File: src/main/java/tk/hiddenname/smarthome/model/task/processing/objects/SetPwmSignalObject.kt
package tk.hiddenname.smarthome.model.task.processing.objects

import com.fasterxml.jackson.databind.PropertyNamingStrategy.SnakeCaseStrategy
import com.fasterxml.jackson.databind.annotation.JsonNaming
import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.Table

@Entity
@Table(name = "processing_set_pwm_signal")
@JsonNaming(SnakeCaseStrategy::class)
data class SetPwmSignalObject(
        @Column(name = "device_id", nullable = false, updatable = false)
        val deviceId: Long = 0,

        @Column(nullable = false)
        val delay: Int = 0,

        @Column(name = "traget_signal", nullable = false)
        val targetSignal: Int = 0
) : ProcessingObject()